dev/tests/integration/testsuite/Magento/Catalog/Block/Product/ListTest.php

```diff
@@ -55,6 +55,10 @@ public function testToolbarCoverage()
         $parent = $this->_getLayout()->createBlock(\Magento\Catalog\Block\Product\ListProduct::class, 'parent');
 
         /* Prepare toolbar block */
+        $this->_getLayout()
+            ->createBlock(\Magento\Catalog\Block\Product\ProductList\Toolbar::class, 'product_list_toolbar');
+        $parent->setToolbarBlockName('product_list_toolbar');
```

Comment:
Why need this logic? According to my understanding existing integration should work without any changes, otherwise existing Magento instances may be broken by this PR.

Reply:
My main reason why I added the above code for toolbar coverage, it was to update with the correct setup for product listing block. The toolbar block needs to be set to a product listing block only from the layout and the test will check if content for the toolbar is not empty. 

The previous implementation used a default toolbar block class, in case this was not provided in the layout, which created the  problem  when you didn't  want to have a toolbar on the product listing. 

The normal way, I think, is to have toolbar declared on layout which in this case you need to have a test that sets the toolbar block in the product listing block and check if it's ok.

I totally agree with you about the tests should work without with any changes, but in my opinion, for this case, the test was used in a situation where the product listing block provided a toolbar block. The product listing will use a toolbar only if it's provided from layout. If you use a default toolbar, we are back to the problem reported, meaning you can not have a product listing without a toolbar block. 

Please let me know your opinion on this.

Reply:
Such change looks backward-incompatible to me for those who previously relied on toolbar block created automatically. Maybe more suitable for `2.3` but let's give somebody else to look into it.

Reply:
Yes, it will create a backward-incompatibility for those who relay on toolbar block created automatically, but I do not see how the toolbar can be removed without creating your custom product listing block where you will be forced to remove it.